activesupport/test/json/encoding_test.rb

```diff
@@ -143,6 +143,13 @@ def test_hash_encoding
     assert_equal %({\"a\":\"b\",\"c\":\"d\"}), sorted_json(ActiveSupport::JSON.encode(:a => :b, :c => :d))
   end
 
+  def test_hash_keys_encoding
+    ActiveSupport.escape_html_entities_in_json = true
+    assert_equal "{\"\\u003c\\u003e\":\"\\u003c\\u003e\"}", ActiveSupport::JSON.encode("<>" => "<>")
+  ensure
+    ActiveSupport.escape_html_entities_in_json = false
```

Comment:
I copied how it's done here: https://github.com/EiNSTeiN-/rails/blob/4-1-stable/activesupport/test/json/encoding_test.rb#L124